EmulatorDetectorProject/EmulatorDetector/src/main/java/net/skoumal/emulatordetector/EmulatorDetector.java

```diff
@@ -26,56 +29,126 @@
      * Detects if app is currenly running on emulator, or real device.
      * @return true for emulator, false for real devices
      */
-    public static boolean isEmulator() {
+    public static boolean deviceIsEmulator() {
+        int rating = 0;
+
+        if (Build.PRODUCT.contains("sdk") ||
+                Build.PRODUCT.contains("Andy") ||
+                Build.PRODUCT.contains("ttVM_Hdragon") ||
+                Build.PRODUCT.contains("google_sdk") ||
+                Build.PRODUCT.contains("Droid4X") ||
+                Build.PRODUCT.contains("nox") ||
+                Build.PRODUCT.contains("sdk_x86") ||
+                Build.PRODUCT.contains("sdk_google") ||
+                Build.PRODUCT.contains("vbox86p")) {
+            rating++;
+        }
 
-        if(rating < 0) { // rating is not calculated yet

```

Comment:
Why do you count rating with every method call? It could take some time to do all the string comparisons and other stuff.